crates/ruff_python_semantic/src/scope.rs

```diff
@@ -70,22 +70,23 @@ impl<'a> Scope<'a> {
     }
 
     /// Returns the ids of all bindings defined in this scope.
-    pub fn binding_ids(&self) -> std::collections::hash_map::Values<&str, BindingId> {
-        self.bindings.values()
+    pub fn binding_ids(&self) -> impl Iterator<Item = BindingId> + '_ {
+        self.bindings.values().copied()
     }
 
     /// Returns a tuple of the name and id of all bindings defined in this scope.
-    pub fn bindings(&self) -> std::collections::hash_map::Iter<&'a str, BindingId> {
-        self.bindings.iter()
+    pub fn bindings(&self) -> impl Iterator<Item = (&&str, BindingId)> + '_ {
+        self.bindings.iter().map(|(name, id)| (name, *id))
```

Comment:
This is a bit tedious. (Should this just be `self.bindings.iter().copied()`? That would change to `&str`.)

Reply:
I think so. `&&str` gets automatically dereferenced by `Rust` to `&str` but using `copied()` is just simpler :)